x-pack/plugins/observability_solution/entity_manager/server/lib/entities/helpers/ingest_pipeline_script_processor_helpers.test.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { initializePathScript, cleanScript } from './ingest_pipeline_script_processor_helpers';
+
+describe('Ingest Pipeline script processor helpers', () => {
+  describe('initializePathScript', () => {
+    it('initializes a single depth field', () => {
+      expect(initializePathScript('someField')).toMatchInlineSnapshot(`
+        "
+
+                if (ctx.someField == null) {
+                    ctx.someField = new HashMap();
+                }
+              "
+      `);
+    });
```

Comment:
If it's a single level field like `tags`, you don't need to initialize the `new HashMap();` 

When you refactored this code you missed where it tests the `currentIndex + 1 === parts.length`. If `parts.length` is `1` and the `currentIndex` is `0` then you're already at the end and you can just assign the value, there is no need to instantiate a `new HashMap()`.